### ISSUE TITLE
Sending Wrong Status and URL to BitBucket

### ISSUE THREAD

**Issue by ph-One** [nishio-dens/bitbucket-pullrequest-builder-plugin#144]
TLDR; Our pull requests are showing build statuses and URLs that point to the wrong Jenkins job. Our job is a Jenkins Pipeline. The installed version of this plugin is 1.4.26, and our Jenkins version is 2.32.3. The trigger is configured using the checkbox in the job config, and not directly in the pipeline script.

----

We've recently (or this is when we noticed it) been plagued with build statuses that are for Jenkins jobs unrelated to the current pull request -- `pull request A` is getting both the build status of `pull request B`, as well as containing the URL of `pull request B`s Jenkins job (when clicking on the build status icon.)

The job is written and managed so that it can run concurrent builds -- we allow one build per Jenkins agent, so if we have 10 agents, that means we can process 10 PRs concurrently.

By manually triggering a build with a, "test this please" comment, we are able to "sync" them back up, most of the time.

----

At times we will see a "rogue" request in BB that updates the build status and URL incorrectly. I have yet to correlate these rogue requests with any build -- they do not match the build times of any of our other jobs.

**Comment by ph-One** [comment]
Issue on our end. Closing.
